osmoutils/accum/accum.go

```diff
@@ -46,6 +46,21 @@ func MakeAccumulator(accumStore store.KVStore, accumName string) error {
 	return nil
 }
 
+// Makes a new accumulator at store/accum/{accumName}
+// Returns error if already exists / theres some overlapping keys
+func MakeAccumulatorWithValueAndShare(accumStore store.KVStore, accumName string, accumValue sdk.DecCoins, totalShares sdk.Dec) error {
```

Comment:
We should ideally have tests for this to ensure it works as intended (e.g. that it actually makes the accum at `store/accum/{accumName}`) – even just a couple sanity tests would be great

Reply:
Good point! Added test cases for this